### DISCUSSION TITLE
Undeletehandler/Soft delete is missing in Serene 6.0.0

### DISCUSSION THREAD

**Discussion by ZioZenigata** [serenity-is/Serenity#6393] [Help] [answered]
Hi,
in demo site (https://serenity.is/demo/Administration/User) , I can delete users but I can't undelete/resume it.
How to implement it?
Thanks
Roberto

**Comment by wirble** [comment, marked as answer]
In all the versions, soft deletes, you have to implement it yourself.  If you have issues, search the issues first, then discussions.  If all else fails, read the docs.  Although the docs is pretty good if you go through tutorial... in any case here is how to implement undelete:
#2741

**Comment by ZioZenigata** [comment]
If it helps anyone this is my code to add to the endpoint.cs for Serene 6.0.0

It's not very elegant but it works


 [HttpPost, AuthorizeDelete(typeof(MyRow))]
        public SaveResponse Undelete(IUnitOfWork uow, UndeleteRequest request, [FromServices] ILanguageSaveHandler handler)
        {            
            uow.Connection.Query(String.Format(@"UPDATE {0} set IsActive=1 WHERE {1}= @Id", MyRow.Fields.TableName, MyRow.Fields.IdField.Name), new { Id = request.EntityId });
            return new SaveResponse { EntityId = request.EntityId   };
        }

**Comment by mehmetakifcolak** [comment]
Create UndeleteHandler.cs
`using MyRequest = Serenity.Services.UndeleteRequest; `
`using MyResponse = Serenity.Services.UndeleteResponse;`
`using MyRow = MyRow;`
`public interface IUndeleteHandler : IUndeleteHandler<MyRow, MyRequest, MyResponse> { }`

    public class UndeleteHandler : UndeleteRequestHandler<MyRow, MyRequest, MyResponse>, IUndeleteHandler
    {
        public UndeleteHandler(IRequestContext context)
            : base(context)
        {
        }
    } 

Add in Endpoint.cs

        [HttpPost, AuthorizeDelete(typeof(MyRow))]
        public UndeleteResponse Undelete(IUnitOfWork uow, UndeleteRequest request,
            [FromServices] IUndeleteHandler handler)
        {
            return handler.Undelete(uow, request);
        }
you should add this in MyRow.cs
`public sealed class MyRow : Row<MyRow.RowFields>, IIdRow, INameRow, IIsActiveRow`
and add in MyGrid.ts
`protected getIsActiveProperty() { return MyRow.isActiveProperty; }`
I hope it helps someone.

**Reply by moseur** [reply]
Hi Mehmet,

When I going to implement your code by creating myundeletehandler.cs then its showing an error as below:

"Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0246	The type or namespace name 'IUndeleteHandler<,,>' could not be found (are you missing a using directive or an assembly reference?)	BigBangERP.Web	E:\BigBang\59-BigBangERP- HRMReorganize - IIsActive-Active\BigBangERP\BigBangERP.Web\Modules\Masters\Brand\RequestHandlers\BrandUndeleteHandler.cs	12	Active"

Can You please help me regarding this issue. 

Thanks in advance

Moseur

**Comment by ZioZenigata** [comment]
Hi Mehmet,
thank you so much for your answer.
Best regards
Roberto

Il giorno ven 26 mag 2023 alle ore 09:47 Mehmet Akif Colak <
***@***.***> ha scritto:
